src/components/Button.tsx

```diff
@@ -0,0 +1,48 @@
+import styled, { css } from "styled-components";
+import { headingSmall } from "components/Typography";
+
+interface ButtonProps {
+  $variant?: "primary" | "secondary";
+}
+
+
+const secondaryStyles = css`
+  background-color: var(--light-text-color);
+  border: 1px solid var(--accent-color);
+  color: var(--accent-color);
+  &:active:not(:disabled) {
+    backgound-color: var(--accent-light-color);
+  }
+`;
+
+export const Button = styled.button<ButtonProps>`
+  ${headingSmall};
+  align-items: center;
+  background-color: var(--accent-color);
+  border: 0px solid transparent;
+  border-radius: 8px;
+  color: var(--light-text-color);
+  cursor: pointer;
+  display: flex;
+  flex-direction: row;
+  gap: .5rem;
+  justify-content: center;
+  margin-bottom: 1.5rem;
+  outline: none;
+  padding: 11px 27px;
+  width: 100%;
+
+  &:active:not(:disabled) {
+    background-color: var(--accent-hover-color);
+    box-shadow: var(--box-shadow-primary);
+  }
+
+  ${({$variant = "primary"}): typeof secondaryStyles | null => {
+    return $variant === "secondary" ? secondaryStyles : null;
+  }}
+
+  &:disabled {
+    opacity: .25;
+    cursor: not-allowed;
+  }
+`;
```

Comment:
```suggestion
const conditionallyAddSecondaryStyle = ({$variant = "primary"}): typeof secondaryStyles | null => {
  return $variant === "secondary" ? secondaryStyles : null;
}

export const Button = styled.button<ButtonProps>`
  ${headingSmall};
  align-items: center;
  background-color: var(--accent-color);
  border: 0px solid transparent;
  border-radius: 8px;
  color: var(--light-text-color);
  cursor: pointer;
  display: flex;
  flex-direction: row;
  gap: .5rem;
  justify-content: center;
  margin-bottom: 1.5rem;
  outline: none;
  padding: 11px 27px;
  width: 100%;

  &:active:not(:disabled) {
    background-color: var(--accent-hover-color);
    box-shadow: var(--box-shadow-primary);
  }

  ${conditionallyAddSecondaryStyle}

  &:disabled {
    opacity: .25;
    cursor: not-allowed;
  }
`;
```

This function would look much better extracted IMO

Reply:
Wow that does look so much better!!

[bitmoji](https://sdk.bitmoji.com/render/panel/8af413ed-694d-413f-93d2-aee2038b1249-55b80ccd-07ab-4f76-b0c1-7ddd60608ffa-v1.png?transparent=1&palette=1&width=246)